src/ui/SearchInterface/SearchInterface.ts

```diff
@@ -51,128 +51,314 @@ export interface ISearchInterfaceOptions {
 }
 
 /**
- * This component is the root and main component of your search interface.<br/>
- * You should place every other component inside this component.<br/>
- * It is also on this component that you call the initialization function.<br/>
- * Since this component is the root of your Search UI, it is recommended that you give it a unique HTML id attribute in order to reference it easily.
+ * The SearchInterface component is the root and main component of your Coveo search interface. You should place all
```

Comment:
Perhaps we could leave the detailed information here for now and eventually remove it and place it in Confluence as you suggest. I'm going to review the Confluence doc soon anyways :)